### PR TITLE
chore: move sample to expo

### LINKED ISSUE #2900: Expo support with react-native-video
I am interested in react-native-video and as I am currently developing with Expo I was just wondering if this component is usable in Expo.

### PR DESCRIPTION
**Pull request by freeboub**
## Summary
move sample to expo

### Motivation
now expo is the standard way to work with react native

### Changes
Add expo support

## Test plan
With sample

### CODE REVIEW

File: examples/basic/package.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
   "version": "1.0.0",
   "private": true,
   "scripts": {
-    "android": "react-native run-android",
-    "ios": "react-native run-ios",
+    "android": "expo run:android",
+    "ios": "expo run:ios",
     "windows": "react-native run-windows",
     "start": "react-native start",
```
Comment (KrzysztofMoch) [inline examples/basic/package.json]: Let's use Expo CLI
```suggestion
    "start": "expo start",
```

File: examples/basic/package.json [hunk 2]
```diff
@@ -14,6 +14,7 @@
   },
   "dependencies": {
     "@react-native-picker/picker": "2.7.5",
+    "expo": "^51.0.0",
```
Comment (KrzysztofMoch) [inline examples/basic/package.json]: Also I think we should update deps (as expo cli says)
```
expo@51.0.10 - expected version: ~51.0.11
react-native@0.74.1 - expected version: 0.74.2
@babel/core@7.23.3 - expected version: ^7.24.0
@types/react@18.2.47 - expected version: ~18.2.79
typescript@5.0.4 - expected version: ~5.3.3
```

**Review by KrzysztofMoch — APPROVED** [review]
Tested, works well ✅ - left two small comments

**Comment by KrzysztofMoch** [comment]
@freeboub, can we merge this ?

**Comment by freeboub** [comment]
@KrzysztofMoch yhere is an issue with local asset I would like to fix (and document) before merging ...

**Comment by freeboub** [comment]
I just retest the branch and it works fine 🤔 So it is OK for merging !

**Comment by freeboub** [comment]
@KrzysztofMoch The crash I found shall be fixed by: https://github.com/TheWidlarzGroup/react-native-video/pull/3915 ...

**Comment by KrzysztofMoch** [comment]
I will merge this as PR with crash fix is merged and CI is passing

**Comment by YangJonghun** [comment]
@freeboub @KrzysztofMoch 
basic example doesn't work API 33 Emulator🤔
(Should clear the cache with `./gradlew clean` to make it reproducible.)

**Pull request merged**